mlrun/artifacts/base.py

```diff
@@ -1067,6 +1068,17 @@ def convert_legacy_artifact_to_new_format(
             f"Unsupported type '{type(legacy_artifact)}' for legacy artifact"
         )
 
+    artifact_key = legacy_artifact_dict.get("key", "")
+    artifact_tag = legacy_artifact_dict.get("tag", "")
+    if artifact_tag:
+        artifact_key = f"{artifact_key}:{artifact_tag}"
+    # TODO: remove in 1.8.0
+    warnings.warn(
+        f"Converting legacy artifact '{artifact_key}' to new format. This will not be supported in MLRun 1.8.0. "
+        f"Make sure to save the artifact/project in the new format.",
```

Comment:
Are you sure saving the artifact will achieve the desired effect?

Reply:
It depends on what the user is doing. It is possible that they loaded the artifact from artifact yaml (import_artifact) and not from project yaml. They will need to save the artifact or the project depending on what they loaded.